### ISSUE TITLE
Support DVD menus

### ISSUE THREAD

**Issue by Igetin** [mpv-player/mpv#14370]
### Expected behavior of the wanted feature

Mpv should support viewing/browsing DVD menus. This should work when reading a disc directly, as well as when opening a `VIDEO_TS` directory. Ideally, both mouse and keyboard navigation should work. For example, VLC supports all these use cases.

---

I am aware of the [wiki FAQ entry](https://github.com/mpv-player/mpv/wiki/FAQ#user-content-Why_is_there_no_DVDBluray_menu_support_And_DVDBD_support_in_general) for DVD menus. The entry was written in 2019, and by a developer no longer associated with the project. The last (now closed) issue about this is also [from over four years ago](https://github.com/mpv-player/mpv/issues/7737). 

So, I'm opening this feature request since there does not already seem to be an open issue to track it. And I'm wondering whether the situation is still the same.

**Comment by Marth64x** [comment]
DVD is a difficult format to work with and the libraries are very old. If a particular menu has value for you, you can actually demux it with ffmpeg now but you will need to know the coordinates of the menu (which I admit is not user-friendly to find or list right now).

`
./mpv/build/mpv --msg-level=ffmpeg=info --demuxer-lavf-o-add=menu=1 --demuxer-lavf-o-add=menu_vts=1 --demuxer-lavf-o-add=pgc=18 "av://dvdvideo:$IN"
`

I have found that outside of using menus to figure out which title is which in a complex DVD, the only practical value they usually add is background music or some gallery features (which you can extract with ffmpeg now)

**Comment by gitproxyid** [comment]
> Does the mpv/FFmpeg thing work with extracted DVDs (just a `VIDEO_TS` folder, no ISO). And how does one figure out the coordinates of the menu?
Yes, it accepts all structures dvdread accepts (VIDEO_TS folder, ISO, actual drive, ...). There are bugfixes and improvements on the way to further improve he demuxer in other ways.
Right now the best way to identify the menu coordinates is to use PgcDemux or PgcEdit, the tools work in WINE.

> My personal issue is the navigation specifically.
If DVD is extracted fully one title at a time, can navigate to chosen segment via filesystem and should never have to deal with the menu again.

**Comment by Marth64x** [comment]
> Does the mpv/FFmpeg thing work with extracted DVDs (just a VIDEO_TS folder, no ISO). And how does one figure out the coordinates of the menu?
Yes, it accepts all structures dvdread accepts (VIDEO_TS folder, ISO, actual drive, ...). There are bugfixes and improvements on the way to further improve he demuxer in other ways.

Right now the best way to identify the menu coordinates is to use PgcDemux or PgcEdit, the tools work in WINE.

>My personal issue is the navigation specifically.
If DVD is extracted to say, MKV, one title at a time, can navigate to chosen segment via filesystem and should never have to deal with the menu again.

**Comment by mlindner** [comment]
> > Does the mpv/FFmpeg thing work with extracted DVDs (just a VIDEO_TS folder, no ISO). And how does one figure out the coordinates of the menu?
> 
> Yes, it accepts all structures dvdread accepts (VIDEO_TS folder, ISO, actual drive, ...). There are bugfixes and improvements on the way to further improve the demuxer in other ways.
> 
> Right now the best way to identify the menu coordinates is to use PgcDemux or PgcEdit, the tools work in WINE.
> 
> > My personal issue is the navigation specifically.
> 
> If DVD is extracted to say, MKV, one title at a time and labeled, one can navigate to chosen segment via filesystem and should never have to deal with the menu again.

Menus are part of the experience of using a bluray format and that should be kept.

**Comment by gitproxyid** [comment]
> Menus are part of the experience of using a bluray format and that should be kept.

Of course they are. But they're a hassle to implement and deal with.

**Comment by Marth64x** [comment]
> Menus are part of the experience of using a bluray format and that should be kept.
Of course they are. But they're a hassle to implement and deal with.

**Comment by llyyr** [comment]
> Menus are part of the experience of using a bluray format and that should be kept.

You're welcome to contribute the code in a way that's not invasive to the core and maintain it

**Comment by kasper93** [comment]
> Menus are part of the experience of using a bluray format and that should be kept.

Of course they are. You can buy standalone blu-ray player for very cheap to get full experience. Menus are interface optimized for certain input method / viewing environment.

As for mpv, patches welcome. We always welcome new contributions, especially for such high requested feature.

> not invasive to the core

This is not fully the only reason it were removed. I've looked at history and wm4 removed way more than necessary, if it were truly code quality issue. Maybe he just didn't like disc formats or had another motives, I don't know. But lots of navigation featured for dvd/bd were removed for no good reason. Not saying full menu support, but just jumping through titles/angles and so on. This could be very easily re-implemented. Like I said contributions are welcome. I'm sure there are people willing to work on this, right?

**Comment by Marth64x** [comment]
>  I'm sure there are people willing to work on this, right?
I am planning to add ability to list playable artifacts (titles, angles, etc.) to ffmpeg demuxer, but it will take some time and has a chance to be rejected. But as seeking is done now, that is my next and probably final feature goal for the demuxer.

**Comment by kasper93** [comment]
> I am planning to add ability to list playable artifacts (titles, angles, etc.) to ffmpeg demuxer

mpv already has own demuxer, that works good and even reads those, but after removal few years back it is not exposed to user control.

**Comment by Baerbeisser** [comment]
There's [libdvdnav](https://www.videolan.org/developers/libdvdnav.html) though. Still a hassle to implement?

**Comment by kasper93** [comment]
> There's [libdvdnav](https://www.videolan.org/developers/libdvdnav.html) though. Still a hassle to implement?

And mpv uses that already for DVD.

**Comment by Timofej559886** [comment]
is there any chance that support for DVD and BD menus will appear?

**Comment by xuhongdi-2026** [comment]
那Arch Linux（CN）社区论坛中苦苦哀求的事，还没有解决吗？
参考链接：https://forum.archlinuxcn.org/t/topic/15982

**Comment by tbertels** [comment]
See https://github.com/mpv-player/mpv/pull/18080.

**Comment by kasper93** [comment]
Probably fixed by: #18080
